apps/passport-client/src/worker/service-worker.ts

```diff
@@ -243,6 +261,16 @@ self.addEventListener("fetch", (event: FetchEvent) => {
         return stableResp;
       }
 
+      // Check Z cache next. We fetch if and only if we do not get a hit.
```

Comment:
It would be a minor optimization, and optional, but with 3 different caches to check, I wonder if it would make more sense for performance to start all the queries up front so they can run in parallel, then only await and check them one at a time.

I'm pretty sure it's safe to abandon the promises if you return early in this case.  (They are simple cache lookups which shouldn't have side-effects, unlike the `startFetch` below which needs special handling to make sure the cache update happens.